programs/ssl/ssl_server2.c

```diff
@@ -1572,13 +1572,15 @@ int main( int argc, char *argv[] )
         else if( strcmp( p, "buffer_size" ) == 0 )
         {
             opt.buffer_size = atoi( q );
-            if( opt.buffer_size < 1 || opt.buffer_size > MBEDTLS_SSL_MAX_CONTENT_LEN + 1 )
+            if( opt.buffer_size < 1 ||
+                    ( opt.buffer_size > MBEDTLS_SSL_IN_CONTENT_LEN + 1
+                      && opt.buffer_size > MBEDTLS_SSL_OUT_CONTENT_LEN ) )
```

Comment:
Question: it's not clear to me why you kept `+1` of the input length but not for the output length. Can you clarify? (Actually it's not clear to me why there was a `+1` here in the first place.)

Reply:
I think the `+1` is put there to allow for an extra `'\0'` used when printing incoming data. Having re-read the code I agree this is also needed for the output length - I'll change this.